controllers/release/release_adapter_test.go

```diff
@@ -220,6 +222,34 @@ var _ = Describe("Release Adapter", Ordered, func() {
 		Expect(k8sClient.Delete(ctx, pipelineRun)).Should(Succeed())
 	})
 
+	It("ensures the target context is set when a workspace is targeted by releaseplan", func() {
+		// Add a target workspace to the ReleasePlan
+		patch := client.MergeFrom(releasePlan.DeepCopy())
+		releasePlan.Spec.Target.Workspace = "test-workspace"
+		Expect(k8sClient.Patch(ctx, releasePlan, patch)).Should(Succeed())
+		Eventually(func() bool {
+			result, err := adapter.EnsureTargetContextIsSet()
+			clusterName, _ := logicalcluster.ClusterFromContext(adapter.targetContext)
+			return !result.RequeueRequest && err == nil &&
+				clusterName.String() == "test-workspace"
```

Comment:
@davidmogar this fails. The reason is that `a.targetContext = logicalcluster.WithCluster(a.context, logicalcluster.New(releasePlan.Spec.Target.Workspace))` fails with `"error": "Index with name cluster does not exist"` basically, AIUI, because the `test-workspace` context doesn't exist, we can't call the `WithCluster` command. Ideas? I looked at `release_suite_test.go` and I am not even sure what the context name is for the default context. The way I see it, we can remove this part of the test (not great) or do all the legwork to setup a fake context and destroy it after this test.

Reply:
I don't even know how you can get that error working exclusively with contexts. Also, you don't need an `Eventually` block for this. It can be safely removed.

In any case, the test fails because when you create the adapter you set the `context` and `targetContext` to the same value. As `targetContext` is not `nil`, the function gets ignored and the cluster name is not set.